cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/KinesisConsumerIT.java

```diff
@@ -0,0 +1,177 @@
+
+        cdcsdkContainer = TestHelper.getCdcsdkContainerForKinesisSink(ybHelper, "public." + DEFAULT_TABLE_NAME)
+                .withCreateContainerCmdModifier(cmd -> cmd.withUser(System.getenv("USERID")));
+        cdcsdkContainer.withNetwork(containerNetwork);
+
+        cdcsdkContainer.withFileSystemBind(System.getenv("AWS_SHARED_CREDENTIALS_FILE"),
+                "/home/jboss/.aws/credentials", BindMode.READ_ONLY);
+        cdcsdkContainer.start();
+
+        assertTrue(cdcsdkContainer.isRunning());
+
+    }
+
+    @AfterEach
+    public void dropTable() throws Exception {
+        ybHelper.execute(UtilStrings.getDropTableStmt(DEFAULT_TABLE_NAME));
+    }
+
+    @Test
+    public void automationOfKinesisAssertions() throws Exception {
+
+        int recordsInserted = 1;
+        String uuid = getUUID().substring(0, 8);
+        for (int i = 0; i < recordsInserted; ++i) {
+            ybHelper.execute(UtilStrings.getInsertStmt(DEFAULT_TABLE_NAME, i, uuid, "last_" + i, 23.45));
+        }
+        ybHelper.execute(UtilStrings.getUpdateStmt(DEFAULT_TABLE_NAME, 0, 25));
+
+        ybHelper.execute(UtilStrings.getDeleteStmt(DEFAULT_TABLE_NAME, 0));
+
+        Awaitility.await().atMost(Duration.ofSeconds(10));
+
+        List<String> expected_data = List.of(
+                "{\"id\":0,\"first_name\":\"" + uuid
+                        + "\",\"last_name\":\"last_0\",\"days_worked\":23.45,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"days_worked\":25.0,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"__deleted\":\"true\"}");
+
+        List<String> actual_data = new ArrayList<>();
+
+        AWSCredentials awsCredentials = new BasicSessionCredentials(System.getenv("AWS_ACCESS_KEY_ID"),
+                System.getenv("AWS_SECRET_ACCESS_KEY"),
+                System.getenv("AWS_SESSION_TOKEN"));
+
+        AmazonKinesisClient client = new AmazonKinesisClient(awsCredentials);
+
+        client.setEndpoint("https://kinesis.ap-south-1.amazonaws.com/");
+
+        final int INTERVAL = 2000;
+        ;
+
+        List<Shard> initialShardData = client.describeStream(stream_name).getStreamDescription().getShards();
+
+        // Getting shardIterators (at beginning sequence number) for reach shard
+        List<String> initialShardIterators = initialShardData.stream()
+                .map(s -> client.getShardIterator(new GetShardIteratorRequest()
+                        .withStreamName(stream_name)
+                        .withShardId(s.getShardId())
+                        .withStartingSequenceNumber(s.getSequenceNumberRange().getStartingSequenceNumber())
+                        .withShardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER))
+                        .getShardIterator())
+                .collect(Collectors.toList());
+
+        String shardIterator = initialShardIterators.get(0);
+
+        // Continuously read data records from a shard
+        while (true) {
```

Comment:
Can you use awaitility here and bound the time to wait for results?

Reply:
We don't have to do that, because getRecords() does not block and will return zero records if there is no data near the part of the shard pointed to by the ShardIterator. In every iteration we update the ShardIterator with NextShardIterator. (https://docs.aws.amazon.com/streams/latest/dev/troubleshooting-consumers.html#getrecords-returns-empty)

Reply:
IIUC, this loop is waiting for the condition `recordResult.getRecords().size() == 0 || shardIterator == null`. Assume the insert fails and only inserts 3 records. In that case, it will loop forever. Instead you'll have to remove `while(true) ` and replace it with the following structure:

    await().until(receivedAllRecords()).atMost(Duration.ofSeconds(INTERVAL));
    
    void receivedAllRecords(...) {
        // code to read from kinesis
    }
    
    
Also add set polling interval using the [polling API](https://github.com/awaitility/awaitility/wiki/Usage#fixed-poll-interval) in awaitility.

Reply:
Done

Reply:
I think you can remove `while(true)`.

Change the end of the function to:

    if (shardIterator == null) throw exception
    
    return validateRecords(...)

Reply:
No the loop is needed, because in each iteration we update the value of shardIterator